frontend/src/app/api/BaseApi.ts

```diff
@@ -69,7 +65,7 @@ export default abstract class BaseApi {
     basePath: string,
     namespace: string,
     subPath: string,
-    queryParamData: QueryParamData,
+    queryParamData?: QueryParamData,
```

Comment:
all queryParamData / searchInput references need to be an optional param because they're only used for search (`SearchOpportunityAPI` class) and not when the derived `OpportunityListingAPI` class is used.